lib/site-inspector.rb

```diff
@@ -127,41 +190,447 @@ def redirect
     end
   end
 
-  def to_json
-    to_hash.to_json
+  def http
+    details = {
+      endpoints: endpoints
+    }
+
+    # convenient shorthand for the extensive statements to come
+    combos = details[:endpoints]
+
+    # A domain is "canonically" at www if:
+    #  * at least one of its www endpoints responds
+    #  * both root endpoints are either down or redirect *somewhere*
+    #  * either both root endpoints are down, *or* at least one
+    #    root endpoint redirect should immediately go to
+    #    an *internal* www endpoint
+    # This is meant to affirm situations like:
+    #   http:// -> https:// -> https://www
+    #   https:// -> http:// -> https://www
+    # and meant to avoid affirming situations like:
+    #   http:// -> http://non-www,
+    #   http://www -> http://non-www
+    # or like:
+    #   https:// -> 200, http:// -> http://www
+
+    www = !!(
```

Comment:
This should be its own method, not a variable in a method. How does this relate to the existing `www?` method? Perhaps `www` should return `true`, `false`, or `:canonical` (which would be truthy as well)?